src/main/scala/main/solutions/MazeSolver/Solver.scala

```diff
@@ -0,0 +1,33 @@
+package main.solutions.MazeSolver
+
+import scala.collection.mutable.{HashSet, Queue}
+
+object Solver {
+  def hasPathBFS(maze: Array[Array[Int]], start: Point, end: Point): Boolean = {
+    hasPathBFSHelper(maze, start, end, HashSet[Point]())
+  }
+  private def hasPathBFSHelper(maze: Array[Array[Int]], start: Point, end: Point, visited: HashSet[Point]): Boolean = {
+    val queue = Queue[Point]()
+    queue.enqueue(start)
+    while (queue.nonEmpty) {
+      val point = queue.dequeue
+      visited.add(point)
+      if (point == end)
+        return true
+      else
+        Helpers.getValidNeighbors(maze, point, visited).foreach(queue.enqueue)
+    }
+    false
+  }
+  def hasPathDFS(maze: Array[Array[Int]], start: Point, end: Point): Boolean = {
+    hasPathDFSHelper(maze, start, end, HashSet[Point]())
+  }
+
+  private def hasPathDFSHelper(maze: Array[Array[Int]], start: Point, end: Point, visited: HashSet[Point]): Boolean = {
+    if (start == end)
+      return true
```

Comment:
no need for "return" statement in Scala, but in this case if you remove it your code won't compile. The point is that in Functional programming there are no side effects, and everything returns a value. so if you'll add "else" after the "if", you'll be able to remove the "return".